src/ops/direnv/mod.rs

```diff
@@ -16,20 +15,19 @@ use std::process::Command;
 pub fn main<W: std::io::Write>(project: Project, mut shell_output: W) -> OpResult {
     check_direnv_version()?;
 
-    let socket_path = crate::ops::get_paths()?.daemon_socket_file().to_owned();
-
     let root_paths = Roots::from_project(&project).paths();
     let paths_are_cached: bool = root_paths.all_exist();
+    let address = crate::ops::get_paths()?.daemon_socket_address();
+    let shell_nix = rpc::ShellNix {
```

Comment:
What if we implemented a TryFrom for ShellNix <-> NixFile? Then we don't need to implement the error checking and conversion time multiple times.

Reply:
Done, nice!